build-logic/src/main/kotlin/ktlint-kotlin-common.gradle.kts

```diff
@@ -5,22 +5,33 @@ plugins {
     kotlin("jvm")
 }
 
+val projectLibs = extensions.getByType<VersionCatalogsExtension>().named("libs")
+val javaCompilationVersion = JavaLanguageVersion.of(projectLibs.findVersion("java-compilation").get().requiredVersion)
+val javaTargetVersion = JavaLanguageVersion.of(projectLibs.findVersion("java-target").get().requiredVersion)
+
 kotlin {
     // All modules, the CLI included, must have an explicit API
     explicitApi()
+    jvmToolchain(jdkVersion = javaCompilationVersion.asInt())
 }
 
+tasks.withType<JavaCompile>().configureEach {
+    options.release.set(javaTargetVersion.asInt())
+}
 tasks.withType<KotlinCompile>().configureEach {
-    compilerOptions {
-        jvmTarget.set(JvmTarget.JVM_1_8)
-    }
+    compilerOptions.jvmTarget.set(JvmTarget.fromTarget(JavaVersion.toVersion(javaTargetVersion).toString()))
```

Comment:
This is the _cost_ of having everything defined at one place - I had to juggle between types :/ 
As an input I may have `"8"` or `"11"` string (representing major java language version) which has to be transformed into kotlin's `JvmTarget` class which expects `"1.8"` or `"11"` string 🤷 The solution I can up with goes through Gradle `JavaVersion` which is able to transform between all variants/types, but maybe there is any better option? 🤷

Reply:
Indeed, not the most pretty one. But with a comment like `Convert Java version (e.g. "1.8" or "11") to Kotlin JvmTarget ("8" resp. "11")` it is clear enough.